### PR TITLE
added life bonds ability from mineplex

### PR DESCRIPTION
**Pull request by ezrn**
-tested

### CODE REVIEW

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        allies.add(player);
+        for (var data : UtilPlayer.getNearbyPlayers(player, distance)) {
+            Player target = data.getKey();
+            boolean friendly = data.getValue() == EntityProperty.FRIENDLY;
+            if (friendly) {
+                allies.add(target);
+            }
+        }
+        return allies;
+    }
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java:130]: Can be replaced with UtilPlayer.getNearbyAllies

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        Location loc = player.getLocation().add(0, 1, 0);
+        player.getWorld().spawnParticle(Particle.CHERRY_LEAVES, loc, 2, 0.2, 0.2, 0.2, 0);
+    }
+
+    private void shareHealth(Player player, int distance) {
+        List<Player> allies = new ArrayList<>();
+        double totalHealth = player.getHealth();
+        allies.add(player);
+
+        for (var data : UtilPlayer.getNearbyPlayers(player, distance)) {
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java:142]: same as above

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        return (float) energy - ((level - 1));
+    }
+
+    private void sendState(Player player, boolean state) {
+        UtilMessage.simpleMessage(player, getClassType().getName(), "Life Bonds: %s", state ? "<green>On" : "<red>Off");
+    }
+
+    @Override
+    public void loadSkillConfig() {
+        minRadius = getConfig("minRadius", 2, Integer.class);
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java:183]: radius should probably be a double?

**Review by Mykindos — CHANGES_REQUESTED** [review]

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java [hunk 4]
```diff
@@ -109,32 +105,26 @@ public void grassAura() {
...
         for (Player ally : allies) {
             spawnParticleAboveHead(ally);
         }
     }
 
-    private List<Player> getAllAllies(Player player, int distance) {
-        List<Player> allies = new ArrayList<>();
+    private List<Player> getAllAllies(Player player, double distance) {
+        List<Player> allies = UtilPlayer.getNearbyAllies(player, player.getLocation(), distance);
         allies.add(player);
```
Comment (Mykindos) [inline champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java:117]: I dont think you need to add yourself to this list, so you could remove this function entirely and just use getNearbyAllies directly

**Review by Mykindos — CHANGES_REQUESTED** [review]

File: champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java (hunk 4 above)
Comment (ezrn) [reply champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java:117]: im pretty sure I use this to do stuff to the caster too so id have to rewrite other parts if I was to remove this

Comment (Mykindos) [reply champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/paladin/passives/LifeBonds.java:117]: Yeah you are right, it filters out the current player by default

**Review by Mykindos — APPROVED** [review]

**Pull request merged**